Library/ViewModels/RewardCardViewModel.swift

```diff
@@ -152,10 +177,65 @@ private func backingReward(fromProject project: Project) -> Reward? {
 
 private func rewardTitle(project: Project, reward: Reward) -> String {
   guard project.personalization.isBacking == true else {
-    return reward.isNoReward
-      ? Strings.Make_a_pledge_without_a_reward()
-      : (reward.title ?? "")
+    return reward.isNoReward ? Strings.Make_a_pledge_without_a_reward() : reward.title.coalesceWith("")
+  }
+
+  if reward.isNoReward {
+    if userIsBacking(reward: reward, inProject: project) {
+      return Strings.Thank_you_for_supporting_this_project()
+    }
+
+    return Strings.Make_a_pledge_without_a_reward()
+  }
+
+  return reward.title.coalesceWith("")
+}
+
+private func pillStrings(project: Project, reward: Reward) -> [String] {
+  var pillStrings: [String] = []
+
+  if let endsAt = reward.endsAt, project.state == .live, endsAt > 0,
+    endsAt >= AppEnvironment.current.dateType.init().timeIntervalSince1970 {
+    let (time, unit) = Format.duration(
+      secondsInUTC: min(endsAt, project.dates.deadline),
```

Comment:
Why would an `endsAt` timestamp ever be greater than the project deadline? 🤔

Reply:
This is just how the code was before... didn't really want to change logic in case it was there for some historical reason.